packages/editor/src/store/selectors.js

```diff
@@ -2209,6 +2210,40 @@ export function getPermalink( state ) {
 	return prefix;
 }
 
+/**
+ * Returns the slug for the post being edited, preferring a manually edited
+ * value if one exists. It falls back to an auto-generated value returned from
+ * an autosave, then to a sanitized version of the current post title, and
+ * finally to the post ID.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {string} The current slug to be displayed in the editor
+ */
+export function getEditedPostSlug( state ) {
+	const currentSlug = getEditedPostAttribute( state, 'slug' );
+
+	if ( currentSlug ) {
+		return currentSlug;
+	}
+
+	// If there is no saved or edited slug and the title has not been edited,
+	// then there may be a php generated slug available on the current_post.
+	// isSaving check is required because the title edit is cleared before the
+	// generated_slug value is set from the response.
+	const { postName } = getPermalinkParts( state );
```

Comment:
There are failures in the E2E suite, which appear legitimate and may stem from this snippet of code.

```
● Sidebar Permalink Panel › should not render link panel when post is publicly queryable but not public
    expect(jest.fn()).not.toHaveErrored(expected)
    Expected mock function not to be called but it was called with:
    ["TypeError: Cannot read property 'postName' of null
```

What are the possible return values of `getPermalinkParts` ? Are we certain it's an object in such a way that we can confidently destructure it?

Reply:
Yep, in this case non public or non publicly queryable posts return null on `getPermalinkParts()` because they don't have a `permalink_template` attribute. But I realized I could get what I need more directly than that since all I really need here is the `generated_slug` post attribute.